uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/leave-request-api_test.js

```diff
@@ -0,0 +1,96 @@
+define([
+  'mocks/data/leave-request-data',
+  'leave-absences/shared/apis/leave-request-api',
+], function (mockData) {
+  'use strict';
+
+  describe('LeaveRequestAPI', function () {
+    var LeaveRequestAPI, $httpBackend;
+
+    beforeEach(module('leave-absences.apis'));
+
+    beforeEach(inject(function (_LeaveRequestAPI_, _$httpBackend_) {
+      LeaveRequestAPI = _LeaveRequestAPI_;
+      $httpBackend = _$httpBackend_;
+
+      //Intercept backend calls for LeaveRequest.all
+      $httpBackend.whenGET(/action\=get&entity\=LeaveRequest/)
+        .respond(function () {
+          return [200, mockData.all()];
+        });
+
+      //Intercept backend calls for LeaveRequest.balanceChangeByAbsenceType
+      $httpBackend.whenGET(/action\=getbalancechangebyabsencetype&entity\=LeaveRequest/)
+        .respond(function () {
+          return [200, mockData.balanceChangeByAbsenceType()];
+        });
+    }));
+
+    it('all() contains expected data', function () {
+      spyOn(LeaveRequestAPI, "getAll").and.callThrough();
+      var leaveRequestPromise = LeaveRequestAPI.all("filters", "pagination", "sort", "params");
```

Comment:
```js
LeaveRequestAPI.all("filters", "pagination", "sort", "params");
```
I don't understand this, this is not representative of how you use the `.all()` method